### PR TITLE
Added YOLOX Tiny model

### PR DESCRIPTION
**Pull request by Anna-Mironova**
- [x] Add demo support
- [x] Update device_support.md

### CODE REVIEW

File: demos/common/python/models/yolo.py [hunk 1]
```diff
@@ -305,3 +305,98 @@ def sigmoid(x):
...
+        preprocessed_image = self.input_transform(padded_image)
+        preprocessed_image = preprocessed_image.transpose((2, 0, 1))  # Change data layout from HWC to CHW
+        preprocessed_image = preprocessed_image.reshape((self.n, self.c, self.h, self.w))
+
+        dict_inputs = {self.image_blob_name: preprocessed_image}
+        return dict_inputs, meta
+
+    def postprocess(self, outputs, meta):
+        if np.size(self.expanded_strides) == 0 or np.size(self.grids) == 0:
+            self.set_strides_grids()
```
Comment (eizamaliev) [inline demos/common/python/models/yolo.py:354]: Why you did it here, but not in ctor?

File: demos/common/python/models/yolo.py [hunk 2]
```diff
@@ -305,3 +305,98 @@ def sigmoid(x):
...
+        image = inputs
+        resized_image = resize_image(image, (self.w, self.h), keep_aspect_ratio=True)
+
+        padded_image = np.ones((self.h, self.w, 3), dtype=np.uint8) * 114
+        padded_image[: resized_image.shape[0], : resized_image.shape[1]] = resized_image
+
+        meta = {'original_shape': image.shape,
+                'resized_shape': resized_image.shape,
+                'padded_shape': padded_image.shape,
+                'scale': min(self.w / image.shape[1], self.h/image.shape[0])}
```
Comment (eizamaliev) [inline demos/common/python/models/yolo.py:343]: You didn't use the `resized_shape` and `padded_shape` in postprocess. Also the `scale` could be obtained directly in postprocess.

File: demos/object_detection_demo/python/object_detection_demo.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def build_argparser():
     args.add_argument('-at', '--architecture_type', help='Required. Specify model\' architecture type.',
                       type=str, required=True, choices=('ssd', 'yolo', 'yolov4', 'faceboxes', 'centernet', 'ctpn',
                                                         'retinaface', 'ultra_lightweight_face_detection',
-                                                        'retinaface-pytorch', 'detr'))
+                                                        'retinaface-pytorch', 'detr', 'yolox'))
```
Comment (eizamaliev) [inline demos/object_detection_demo/python/object_detection_demo.py:52]: Please, group YOLO* models together. Also, don't forget the readme file.

**Review by eizamaliev — CHANGES_REQUESTED** [review]

File: models/public/yolox-tiny/model.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    in_channels = [256, 512, 1024]
+    num_classes = 80
+    depth = 0.33
+    width = 0.375
+    backbone = YOLOPAFPN(depth, width, in_channels=in_channels)
+    head = YOLOXHead(num_classes, width, in_channels=in_channels)
+    model = YOLOX(backbone, head)
+
+    model.apply(init_yolo)
+    model.head.initialize_biases(1e-2)
```
Comment (eaidova) [inline models/public/yolox-tiny/model.py:33]: do we really need weights initialization if you load checkpoint?

File: models/public/yolox-tiny/accuracy-check.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    datasets:
+      - name: ms_coco_detection_80_class_without_background
+
+        preprocessing:
+          - type: resize
+            aspect_ratio_scale: fit_to_window
+            size: 416
+          - type: padding
+            pad_type: right_bottom
+            pad_value: '114, 114, 114'
```
Comment (eaidova) [inline models/public/yolox-tiny/accuracy-check.yml:55]: strange number for padding, are you sure that it is correct?

Comment (Anna-Mironova) [reply models/public/yolox-tiny/accuracy-check.yml:55]: This value is taken from the `preproc` function from the original repository https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/data/data_augment.py#L153.

File: demos/common/python/models/yolo.py (hunk 1 above)
Comment (Anna-Mironova) [reply demos/common/python/models/yolo.py]: Moved

File: demos/common/python/models/yolo.py (hunk 2 above)
Comment (Anna-Mironova) [reply demos/common/python/models/yolo.py]: Done

File: demos/object_detection_demo/python/object_detection_demo.py (hunk 3 above)
Comment (Anna-Mironova) [reply demos/object_detection_demo/python/object_detection_demo.py]: Done

File: models/public/yolox-tiny/model.py (hunk 4 above)
Comment (Anna-Mironova) [reply models/public/yolox-tiny/model.py]: Done

Comment (eaidova) [reply models/public/yolox-tiny/model.py]: not all, model.apply is also weights initialization

**Review by eizamaliev — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

File: models/public/yolox-tiny/model.py (hunk 4 above)
Comment (Anna-Mironova) [reply models/public/yolox-tiny/model.py]: If model is converted without model.apply, we get a decrease in accuracy of about 0.5-1.1 % depending on the metric.

**Review by Wovchena — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
@nignatovsky PR merged, but seems pre-commit check has not catch issue with documentation, and this affect other PRs, like https://github.com/openvinotoolkit/open_model_zoo/pull/2723 - 14:16:05 models/public/yolox-tiny/README.md: URL reference "../../../tools/downloader/README.md" target does not exist or is not a file

cc @jkamelin @Wovchena

**Pull request merged**